gitlabform/processors/group/group_settings_processor.py

```diff
@@ -1,12 +1,30 @@
+from logging import info, debug
+from typing import Dict
+
 from gitlabform.gitlab import GitLab
-from gitlabform.processors.single_entity_processor import SingleEntityProcessor
+from gitlab.v4.objects.groups import Group
+from gitlabform.processors.abstract_processor import AbstractProcessor
 
 
-class GroupSettingsProcessor(SingleEntityProcessor):
+class GroupSettingsProcessor(AbstractProcessor):
     def __init__(self, gitlab: GitLab):
-        super().__init__(
-            "group_settings",
-            gitlab,
-            get_method_name="get_group_settings",
-            edit_method_name="put_group_settings",
-        )
+        super().__init__("group_settings", gitlab)
+
+    def _process_configuration(self, group: str, configuration: Dict):
+        configured_group_settings = configuration.get("group_settings", {})
+
+        gitlab_group: Group = self.gl.get_group_by_path_cached(group)
+
+        if self._needs_update(gitlab_group.asdict(), configured_group_settings):
+            info("Updating group_settings  '%s'", gitlab_group.name)
```

Comment:
Nit: we should use only the f-strings syntax everywhere in the new code. It's the cleanest. :)